### ISSUE TITLE
Iceberg `rollback_to_snapshot` procedure fails when the snapshots are over 1 minute from each other

### ISSUE THREAD

**Issue by findinpath** [trinodb/trino#12353]
```
trino> USE iceberg.default

trino:default> create table t2 (a int);
CREATE TABLE
trino:default> insert into t2 values (1);
INSERT: 1 row

Query 20220512_141535_00045_7ew9z, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220512_141535_00045_7ew9z
Splits: 19 total, 19 done (100.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 2% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 530B
1.48 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> insert into t2 values (2);
INSERT: 1 row

Query 20220512_141606_00046_7ew9z, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220512_141606_00046_7ew9z
Splits: 19 total, 19 done (100.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 2% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 530B
1.61 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> insert into t2 values (3);
INSERT: 1 row

Query 20220512_141807_00047_7ew9z, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220512_141807_00047_7ew9z
Splits: 19 total, 19 done (100.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 2% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 530B
2.04 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> select * from "t2$snapshots";
             committed_at              |     snapshot_id     |      parent_id      | operation |                                                         manifest_list                                                          |                             >
---------------------------------------+---------------------+---------------------+-----------+--------------------------------------------------------------------------------------------------------------------------------+----------------------------->
 2022-05-12 16:15:22.276 Europe/Vienna | 1491543011198750951 |                NULL | append    | hdfs://hadoop-master:9000/user/hive/warehouse/t2/metadata/snap-1491543011198750951-1-a10948a4-5840-40f0-826c-b82fc342953c.avro | {changed-partition-count=0, >
 2022-05-12 16:15:36.069 Europe/Vienna | 9073277896198277458 | 1491543011198750951 | append    | hdfs://hadoop-master:9000/user/hive/warehouse/t2/metadata/snap-9073277896198277458-1-9567d726-5197-4ca7-b50d-a296dff1cd5d.avro | {changed-partition-count=1, >
 2022-05-12 16:16:07.215 Europe/Vienna | 3763854319464927592 | 9073277896198277458 | append    | hdfs://hadoop-master:9000/user/hive/warehouse/t2/metadata/snap-3763854319464927592-1-53a085c0-7154-47b4-8bd2-185a102e95d8.avro | {changed-partition-count=1, >
 2022-05-12 16:18:08.269 Europe/Vienna |  314706806653167660 | 3763854319464927592 | append    | hdfs://hadoop-master:9000/user/hive/warehouse/t2/metadata/snap-314706806653167660-1-3f80a91b-bc20-415e-b764-40692f468bd1.avro  | {changed-partition-count=1, >
(4 rows)

Query 20220512_141820_00048_7ew9z, FINISHED, 1 node
http://localhost:8080/ui/query.html?20220512_141820_00048_7ew9z
Splits: 9 total, 9 done (100.00%)
CPU Time: 0.0s total,   666 rows/s,  336KB/s, 20% active
Per Node: 0.0 parallelism,     6 rows/s, 3.05KB/s
Parallelism: 0.0
Peak Memory: 4.4KB
0.66 [4 rows, 2.02KB] [6 rows/s, 3.05KB/s]

trino:default> call system.rollback_to_snapshot('default', 't2', 9073277896198277458);
Query 20220512_141847_00049_7ew9z failed: [BUG] Expected sorted snapshot log entries.
io.trino.spi.TrinoException: [BUG] Expected sorted snapshot log entries.
	at io.trino.execution.CallTask.execute(CallTask.java:216)
	at io.trino.execution.CallTask.execute(CallTask.java:71)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:145)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:243)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_dev____20220512_134519_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.IllegalArgumentException: [BUG] Expected sorted snapshot log entries.
	at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument(Preconditions.java:142)
	at org.apache.iceberg.TableMetadata.<init>(TableMetadata.java:306)
	at org.apache.iceberg.TableMetadata$Builder.build(TableMetadata.java:1040)
	at org.apache.iceberg.TableMetadata.replaceCurrentSnapshot(TableMetadata.java:501)
	at org.apache.iceberg.SnapshotProducer.lambda$commit$2(SnapshotProducer.java:289)
	at org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:404)
	at org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:214)
	at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:198)
	at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:190)
	at org.apache.iceberg.SnapshotProducer.commit(SnapshotProducer.java:282)
	at org.apache.iceberg.SnapshotManager.commit(SnapshotManager.java:32)
	at io.trino.plugin.iceberg.RollbackToSnapshotProcedure.rollbackToSnapshot(RollbackToSnapshotProcedure.java:69)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:710)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
	at io.trino.execution.CallTask.execute(CallTask.java:209)
	... 8 more
```

**Comment by bitsondatadev** [comment]
Good find @findinpath, I was going through an older tutorial and ran into this one.

**Comment by findinpath** [comment]
Do note that as of `386` Trino uses Iceberg `0.13.2` which means that this problem should have been solved in the meantime.

However, I am encountering a different issue now:

```
trino:default> create table t2 (a integer);
CREATE TABLE
trino:default> insert into t2 values 1;
INSERT: 1 row

Query 20220617_054905_00010_qdyhx, FINISHED, 1 node
Splits: 19 total, 19 done (100.00%)
1.39 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> insert into t2 values 2;
INSERT: 1 row

Query 20220617_055212_00011_qdyhx, FINISHED, 1 node
Splits: 19 total, 19 done (100.00%)
1.60 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> select * from "t2$snapshots";
             committed_at              |     snapshot_id     |      parent_id      | operation |                              >
---------------------------------------+---------------------+---------------------+-----------+------------------------------>
 2022-06-17 07:48:57.359 Europe/Vienna | 8877014715680553605 |                NULL | append    | hdfs://hadoop-master:9000/use>
 2022-06-17 07:49:06.499 Europe/Vienna |  546732699615194599 | 8877014715680553605 | append    | hdfs://hadoop-master:9000/use>
 2022-06-17 07:52:12.865 Europe/Vienna | 5328766858763350946 |  546732699615194599 | append    | hdfs://hadoop-master:9000/use>
(3 rows)

Query 20220617_055221_00012_qdyhx, FINISHED, 1 node
Splits: 9 total, 9 done (100.00%)
0.30 [3 rows, 1.49KB] [10 rows/s, 4.99KB/s]

trino:default> call system.rollback_to_snapshot('default', 't2', 546732699615194599);
Query 20220617_055250_00013_qdyhx failed: During instantiation, the Configuration object captures the TCCL and uses it to resolve classes by name. For this reason, the current TCCL jdk.internal.loader.ClassLoaders$AppClassLoader@77556fd should be same as this class's classloader io.trino.server.PluginClassLoader@5e8e40e8. Otherwise the constructed Configuration will use *some* classloader to resolve classes
```

This is related to https://github.com/trinodb/trino/pull/12687

I will create a different issue for this one.

**Comment by findinpath** [comment]
> @findinpath did you verify this is solved?

I did verify whether the problem is still occurring, after the iceberg library update. However I stumbled on the issue reported at #12887 . We should verify this functionality again to be sure that this is indeed fixed once that #12887 is fixed. Thank you for raising this point.
